### PR TITLE
Add compass integration jobs

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- Add compass pre-submit, post-submit and release integration jobs

**Related issue(s)**

kyma-project/kyma#4685

### CODE REVIEW

File: development/tools/jobs/incubator/compass/compass_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/kyma-project/test-infra/development/tools/jobs/tester"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const compassIntegrationTestJobPath = "./../../../../../prow/jobs/incubator/compass/compass-integration.yaml"
+
+func TestCompassIntegrationVMJobsReleases(t *testing.T) {
+	// WHEN
+	unsupportedReleases := []tester.SupportedRelease{tester.Release11}
```
Comment (pkosiec) [inline development/tools/jobs/incubator/compass/compass_integration_test.go]: Exclude 1.2

File: development/tools/jobs/incubator/compass/compass_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+func TestCompassIntegrationJobsPresubmit(t *testing.T) {
+	tests := map[string]struct {
+		givenJobName            string
+		expPresets              []tester.Preset
+		expRunIfChangedRegex    string
+		expRunIfChangedPaths    []string
+		expNotRunIfChangedPaths []string
+	}{
+		"Should contains the compass-integration job": {
```
Comment (pkosiec) [inline development/tools/jobs/incubator/compass/compass_integration_test.go]: `Should contain` without `s`

image

File: development/tools/jobs/incubator/compass/compass_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		})
+	}
+}
+
+func TestKymaIntegrationJobsPostsubmit(t *testing.T) {
+	tests := map[string]struct {
+		givenJobName string
+		expPresets   []tester.Preset
+	}{
+		"Should contains the compass-integration job": {
```
Comment (pkosiec) [inline development/tools/jobs/incubator/compass/compass_integration_test.go]: contain without s

File: prow/jobs/incubator/compass/compass-integration.yaml [hunk 4]
```diff
@@ -39,13 +37,13 @@ presubmits: # runs on PRs
     - ^master$
     <<: *vm_job_template
     # following regexp won't start build if only Markdown files were changed
-    # run_if_changed: "^((chart/\\S+|installation/\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))" # TODO: Uncomment when done testing
+    run_if_changed: "^((chart/\\S+|installation/\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))"
```
Comment (pkosiec) [inline prow/jobs/incubator/compass/compass-integration.yaml]: Can we simplify it?

Comment (tgorgol) [reply prow/jobs/incubator/compass/compass-integration.yaml]: Hard to simplify it without negative lookahead but I found some edge cases in that regex and fixed them.

File: prow/jobs/incubator/compass/compass-integration.yaml [hunk 5]
```diff
@@ -39,27 +37,15 @@ presubmits: # runs on PRs
     - ^master$
     <<: *vm_job_template
     # following regexp won't start build if only Markdown files were changed
-    # run_if_changed: "^((chart/\\S+|installation/\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))" # TODO: Uncomment when done testing
+    run_if_changed: "^(chart/\\S+|installation/\\S+)\\.([^.\\s]|[^.\\s]{3,}|[^.\\s][^dD]|[^mM][^.\\s])$"
```
Comment (aszecowka) [inline prow/jobs/incubator/compass/compass-integration.yaml]: please simplify this regexp - IMO it is better to have simple regexp even if sometimes we run integration tests if we modified only documentation files

**Pull request merged**